content/doc/developer/extensibility/index.adoc

```diff
@@ -22,4 +22,30 @@ references:
   title: My class is missing descriptor
 - url: https://wiki.jenkins-ci.org/display/JENKINS/Extension+Point+for+Project+Views+Navigation
   title: Extension Point for Project Views Navigation # TODO make into guide or remove as obsolete with BO
+- url: https://github.com/jenkinsci/sample-extensions
+  title: Sample extension implementations
 ---
+
+Jenkins is designed with extensibility in mind, both internally (core) and with plugins.
+Extensibility in Jenkins is accomplished by combining the concepts discussed below.
+
+== Extension annotation
```

Comment:
Add: `@Initializer` to the "Initializers" section. 
`MyWeirdThing extends Plugin` is probably an extensibility thing as well, but I'd guess it is more or less deprecated now